From ae5b1c4df1a48f275f78f1d08de9462c6aec886f Mon Sep 17 00:00:00 2001
From: Rocky Hao <rocky.hao@rock-chips.com>
Date: Mon, 5 Feb 2018 19:00:48 +0800
Subject: [PATCH] arm: dts: rockchip: add dmc thermal control for rk322x

add power model for dmc and add dmc as a cooling device in thermal
control

Change-Id: Ibc5f8ec97b43d4fcf9ff817582711b924d2a7ef0
Signed-off-by: Rocky Hao <rocky.hao@rock-chips.com>
---
 arch/arm/boot/dts/rk322x.dtsi | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/boot/dts/rk322x.dtsi b/arch/arm/boot/dts/rk322x.dtsi
index d03bf7c656d5..790226c01977 100644
--- a/arch/arm/boot/dts/rk322x.dtsi
+++ b/arch/arm/boot/dts/rk322x.dtsi
@@ -173,7 +173,16 @@
 		>;
 		dram_freq = <786000000>;
 		rockchip,dram_timing = <&dram_timing>;
+		#cooling-cells = <2>;
 		status = "disabled";
+
+		ddr_power_model: ddr_power_model {
+			compatible = "ddr_power_model";
+			dynamic-power-coefficient = <120>;
+			static-power-coefficient = <200>;
+			ts = <32000 4700 (-80) 2>;
+			thermal-zone = "soc-thermal";
+		};
 	};
 
 	dmc_opp_table: dmc-opp-table {
@@ -617,6 +626,12 @@
 					<&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
 					contribution = <1024>;
 				};
+				map2 {
+					trip = <&target>;
+					cooling-device =
+					<&dmc THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+					contribution = <1024>;
+				};
 			};
 		};
 	};
-- 
2.35.3

